### PR TITLE
Faculty view

### PR DESCRIPTION
**Pull request by Princekumarofficial**
- [x] #27 #36 
- [x] Faculty view Done
- [x] Feedback Form Done
- [x] API integrated
- [x] ColumnMapping changed such that it shows N/A for none values and shows more nested dto
- [x] Made Table Component responsive

Screenshot 2024-05-10 195644

Screenshot 2024-05-10 195658

Screenshot 2024-04-08 152801

Screenshot 2024-05-10 201151

Screenshot 2024-05-10 201211

### CODE REVIEW

File: src/app/(routes)/faculty/[facultyId]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
+"use client";
+import { fetchApprovals, updateApproval } from "@/helpers/api";
+import Cookies from "js-cookie";
+import TableComponent from "@/components/TableComponent/TableComponent";
+import generateColumns from "@/components/TableComponent/ColumnMapping";
+import { useState, useEffect } from "react";
+
+const dto = [
```
Comment (mittal-ishaan) [inline src/app/(routes)/faculty/[facultyId]/page.tsx:8]: can we change this to
```
status
remarks
salary
     Salary Period
     total CTC
job
     role
     company
         name
     season
         year
         type
         ```
        the backend will now send this.

**Comment by mittal-ishaan** [comment]
@Princekumarofficial the can you do small changes. the backend will now send data according to https://github.com/Web-Team-IITI-Gymkhana/tpc-backend/issues/82

File: src/app/(routes)/faculty/[facultyId]/page.tsx (hunk 1 above)
Comment (Princekumarofficial) [reply src/app/(routes)/faculty/[facultyId]/page.tsx:8]: @mittal-ishaan Done changed the dto

File: src/helpers/api.ts [hunk 2]
```diff
@@ -142,3 +152,37 @@ export const fetchJobEvents = async (
...
+    }
+  );
+  const json = await res.json();
+  return json;
+};
+
+export async function updateApproval(data: any[]) {
+  fetch(url("/faculty-approval"), {
+    method: "PATCH",
+    headers: { "Content-type": "application/json" },
```
Comment (web-team-iiti) [inline src/helpers/api.ts:180]: auth header needed here?

File: src/helpers/api.ts [hunk 3]
```diff
@@ -142,3 +152,37 @@ export const fetchJobEvents = async (
...
+        Authorization: `Bearer ${accessToken}`,
+      },
+    }
+  );
+  const json = await res.json();
+  return json;
+};
+
+export async function updateApproval(data: any[]) {
+  fetch(url("/faculty-approval"), {
```
Comment (web-team-iiti) [inline src/helpers/api.ts:178]:
```suggestion
  fetch(url("/approvals"), {
```

File: src/helpers/api.ts [hunk 4]
```diff
@@ -142,3 +152,37 @@ export const fetchJobEvents = async (
...
+export const fetchApprovals = async (
+  accessToken: string | undefined,
+  filter: string | undefined
+) => {
+  if (!accessToken || accessToken === undefined) {
+    redirect();
+    return;
+  }
+  const res = await fetch(
+    filter ? url(`/faculty-approval?${filter}`) : url("/faculty-approval"),
```
Comment (web-team-iiti) [inline src/helpers/api.ts:165]:
```suggestion
    filter ? url(`/approvals?${filter}`) : url("/approvals"),
```

**Comment by Princekumarofficial** [comment]
Updated DTO

Screenshot 2024-06-12 221606

**Comment by AryanGKulkarni** [comment]
Make a profile page where he can view and update details about himself

File: src/app/(routes)/faculty/[facultyId]/page.tsx [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+const dynamicColumns = generateColumns(dto);
+
+const FacultyPage = () => {
+  const [data, setData] = useState(null);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const jsonData = await fetchApprovals(undefined, undefined);
```
Comment (AryanGKulkarni) [inline src/app/(routes)/faculty/[facultyId]/page.tsx:55]: why undefined?

File: src/helpers/api.ts
Comment (AryanGKulkarni) [inline src/helpers/api.ts:1]: Make a separate folder in helpers named faculty and move the apis in that folder

File: src/app/(routes)/faculty/[facultyId]/page.tsx [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+import { fetchApprovals, updateApproval } from "@/helpers/api";
+import Cookies from "js-cookie";
+import TableComponent from "@/components/TableComponent/TableComponent";
+import generateColumns from "@/components/TableComponent/ColumnMapping";
+import { useState, useEffect } from "react";
+import loadingImg from "@/components/Faculty/Spinner@1x-1.0s-200px-200px (1).svg";
+
+const dto = [
+  {
+    // id: "string",
```
Comment (mittal-ishaan) [inline src/app/(routes)/faculty/[facultyId]/page.tsx:11]: remove commented code if not in use

File: src/components/Faculty/Spinner@1x-1.0s-200px-200px (1).svg
Comment (mittal-ishaan) [inline src/components/Faculty/Spinner@1x-1.0s-200px-200px (1).svg:1]: rename it for better readability

File: src/components/TableComponent/TableComponent.tsx [hunk 7]
```diff
@@ -141,115 +143,126 @@ export default function TableComponent({
...
-              <ImportData />
-              <DropdownMenuSeparator className="border-gray-300 mx-2 bg-gray-300 my-2" />
-              <ExportData data={tableData} />
-              <DropdownMenuSeparator className="border-gray-300 mx-2 bg-gray-300 my-2" />
               <Button
-                onClick={() => {
-                  copyIDsToClipboard(table);
+                onClick={async () => {
+                  setFilters([]);
+                  // const AllStudents = await fetchStudentData(
```
Comment (mittal-ishaan) [inline src/components/TableComponent/TableComponent.tsx:169]: remove if not in use.

File: src/helpers/api.ts [hunk 8]
```diff
@@ -51,25 +55,31 @@ export const fetchAllJobs = async (accessToken: string | undefined,filter: strin
...
     return;
   }
-  console.log('filter', filter)
-  const res = await fetch(filter ? url(`/students?${filter}`) : url("/students"), {
-    next: { tags: ["AllStudents"] },
-    headers: {
-      Authorization: `Bearer ${accessToken}`,
-    },
-  });
+  console.log("filter", filter);
```
Comment (mittal-ishaan) [inline src/helpers/api.ts:66]: remove console log

File: src/helpers/api.ts [hunk 9]
```diff
@@ -34,12 +35,15 @@ export const fetchCompany = async (accessToken: string | undefined) => {
...
+export const fetchAllJobs = async (
+  accessToken: string | undefined,
+  filter: string | undefined
+) => {
   if (!accessToken || accessToken === undefined) {
     redirect();
     return;
   }
-  console.log('filter', filter)
+  console.log("filter", filter);
```
Comment (mittal-ishaan) [inline src/helpers/api.ts:46]: remove console log

File: src/components/Faculty/FeedbackFrom.tsx [hunk 10]
```diff
@@ -0,0 +1,207 @@
...
+      variant={"default"}
+      className="bg-sky-600 mx-4 hover:bg-blue-600"
+      onClick={() => {
+        if (finalButton) {
+          UpdateApprovalPATCH(rows, remarks, "APPROVED");
+        }
+      }}
+    >
+      Approve
+      <svg
```
Comment (mittal-ishaan) [inline src/components/Faculty/FeedbackFrom.tsx:51]: we might want to have svg in a component maybe and use it. the code might be cleaner. though your choice

Comment (Princekumarofficial) [reply src/components/Faculty/FeedbackFrom.tsx:51]: Ok

File: src/app/(routes)/faculty/[facultyId]/page.tsx
Comment (AryanGKulkarni) [inline src/app/(routes)/faculty/[facultyId]/page.tsx:1]: Remove slug we don't need it
we are using access token to identify the faculty

**Review by AryanGKulkarni — CHANGES_REQUESTED** [review]

**Comment by Princekumarofficial** [comment]
Screenshot 2024-06-16 161224
Screenshot 2024-06-16 161228

**Review by AryanGKulkarni — APPROVED** [review]

**Pull request merged**